packages/website/pages/blog/post/[slug].js

```diff
@@ -0,0 +1,302 @@
+    ? files
+        .filter(filename => !filename.startsWith('.'))
+        .map((fn, index) => {
+          const content = fs.readFileSync(`posts/${fn}`).toString();
+          const info = matter(content);
+          return {
+            ...info.data,
+            slug: fn.split('.')[0],
+          };
+        })
+    : [];
+
+  return {
+    props: {
+      posts,
+      post: post,
+      title: data.title + ' - Web3.Storage',
+      image: data.thumbnail,
+      description: data.description,
+    },
+  };
+}
+
+/**
+ * Related Posts
+ */
+const RelatedPosts = ({ items }) => (
+  <>
+    {items.map((post, i) => (
+      <Card key={i} post={post} />
+    ))}
+  </>
+);
+
+/**
+ * Blog Post Page
+ *
+ * @param {Object} props
+ * @returns {JSX.Element}
+ */
+const Post = ({ post, posts }) => {
+  const MainContent = dynamic(() =>
+    import(`/posts/${post.meta.slug}.mdx`).then(component => {
+      // Client-side post-render javascript functionality
+      if (typeof window === 'object') {
+        requestAnimationFrame(() => {
+          CodeHighlightCopy('.post-content pre');
+          // open external links in new tab
+          const externalLinks = document.querySelectorAll('.post-content a[href*="://"]');
+          externalLinks?.forEach(link => {
+            // @ts-ignore
+            link.getAttribute('href') && link.hostname !== window.location.hostname && (link.target = '_blank');
+          });
+        });
+      }
+      return component;
+    })
+  );
+
+  const [showCopied, setShowCopied] = useState(false);
+  const sections = BlogArticlePageData.page_content;
+  const animations = BlogArticlePageData.floater_animations;
+
+  /**
+   * share links
+   */
+  const [currentUrl, setCurrentUrl] = useState('');
+  useEffect(() => setCurrentUrl(window.location.href), []);
+  const twitterShareLink = new URL('https://twitter.com/intent/tweet');
+  const twitterParams = {
+    url: currentUrl,
+    text: '',
+    hashtags: 'web3.storage',
+  };
+  const facebookShareLink = new URL('https://www.facebook.com/sharer/sharer.php');
+  const facebookParams = {
+    u: currentUrl,
+    quote: '',
+    hashtag: '#web3.storage',
+  };
+  const linkedinShareLink = new URL('https://www.linkedin.com/sharing/share-offsite');
+  const linkedinParams = { url: currentUrl };
+
+  /**
+   * get next and previous slugs
+   */
+  const { prevPostSlug, nextPostSlug } = useMemo(() => {
+    const currentPostIndex = posts.findIndex(targetPost => isEqual(targetPost, post.meta));
+    return {
+      prevPostSlug: posts[currentPostIndex - 1]?.slug,
+      nextPostSlug: posts[currentPostIndex + 1]?.slug,
+    };
+  }, [posts, post]);
+
+  /*
+   * Related Posts
+   *
+   * Add in most recent posts of same category first
+   * If related posts < 4 still, add in posts with matching tags (sorted by most tags which match first)
+   * If related posts < 4 still, add in posts at random from the remaining pool
+   */
+  const relatedPosts = useMemo(() => {
```

Comment:
This is a pretty big file, could we split out some of these into separate files to clean it up a bit?